util/kustomize/kustomize.go

```diff
@@ -63,6 +79,67 @@ func (k *kustomize) Build(opts KustomizeBuildOpts, overrides []*v1alpha1.Compone
 
 	return objs, append(getImageParameters(objs)), nil
 }
+func (k *kustomize) GetCommandName() (string, error) {
+
+	version, err := k.getKustomizationVersion()
+	if err != nil {
+		return "", err
+	}
+
+	log.Infof("using kustomize version=%d", version)
+
+	if version == 1 {
+		return "kustomize", nil
+	} else {
+		return "kustomize" + strconv.Itoa(version), nil
+	}
+}
+
+var kustomisations = []string{"kustomization.yaml", "kustomization.yml", "Kustomization"};
+
+// kustomization is a file called kustomization.yaml that describes a configuration consumable by kustomize.
+func (k *kustomize) findKustomization() (string, error) {
+	for _, file := range kustomisations {
+		kustomization := filepath.Join(k.path, file)
+		log.Infof("path=%s, file=%s", k.path, file)
+		if _, err := os.Stat(kustomization); err == nil {
+			return kustomization, nil
+		}
+	}
+	return "", errors.New("did not find kustomization in " + k.path)
+}
+
+func IsKustomization(path string) bool {
+	return path == "kustomization.yaml" || path == "kustomization.yml" || path == "Kustomization"
+}
+
+func (k *kustomize) getKustomizationVersion() (int, error) {
+
+	kustomization, err := k.findKustomization()
+	if err != nil {
+		return 0, err
+	}
+
+	log.Infof("using kustomization=%s", kustomization)
+
+	file, err := os.Open(kustomization)
+	if err != nil {
+		return 0, err
+	}
+
+	defer util.Close(file)
+
+	scanner := bufio.NewScanner(file)
+
+	for scanner.Scan() {
+		// +apiVersion: kustomize.config.k8s.io/v1beta1
+		// +kind: Kustomization
+		if strings.Contains(scanner.Text(), "kind: Kustomization") {
```

Comment:
This check will be sensitive to extra spaces. You may want to do something like:
```golang
type kustomization struct {
    kind string
}
dat, err := ioutil.ReadFile(kustomization)
if err != nil {
    return 0, err
}
var obj kustomization
err = yaml.Unmarshal(dat, &obj}
if err != nil {
    return 0, err
}
if obj.kind == "Kustomization" {
    return 2, nil
}
return 1, nil
```